mockwebserver/src/main/kotlin/mockwebserver3/internal/util/util.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package mockwebserver3.internal.util
+
+import java.net.Inet6Address
+import java.net.InetAddress
+
+/**
+ * Equivalent of InetAddress.getHostName() but with some tweaks for OkHttp behaviour,
+ * assume 127.0.0.1 is localhost and square bracket encode IPv6 addresses.
+ */
+fun InetAddress.quickHostname(): String {
```

Comment:
I wonder if there's a better word for this. I think the square brackets treatment is URL-specific so maybe urlHostname?

This should be a val, not a fun?